### ISSUE TITLE
Does it work with browserify/CommonJS?

### ISSUE THREAD

**Issue by OnkelTem** [AnalyticalGraphicsInc/cesium#4417]
I know it should work but for some reason it doesn't for my setup.  Steps to reproduce:

1) install yeoman and yeogurt generator:
```
sudo npm install -g yeoman
sudo npm install -g generator-yeogurt
```
2) start new project:
```
yo yeogurt
```
3) install cesium:
```
npm install cesium
```
4) add `var Cesium = require('cesium');` to the `src/_scripts/main.js`
5) run `gulp build`:
```
[04:12:47] Browserify compile error: 
 Error: Cannot find module 'lang' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:46:17
    at process (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:173:43)
    at ondir (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:188:17)
    at load (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:69:43)
    at onex (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:92:31)
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15) 

[04:12:47] gulp-notify: [Error running Gulp] Cannot find module 'lang' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
./src/_scripts/main.js was browserified: 1.968s
[04:12:47] Browserify compile error: 
 Error: Cannot find module 'prim' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:46:17
    at process (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:173:43)
    at ondir (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:188:17)
    at load (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:69:43)
    at onex (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:92:31)
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15) 

[04:12:47] gulp-notify: [Error running Gulp] Cannot find module 'prim' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
[04:12:47] Browserify compile error: 
 Error: Cannot find module 'logger' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:46:17
    at process (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:173:43)
    at ondir (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:188:17)
    at load (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:69:43)
    at onex (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:92:31)
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15) 

[04:12:47] gulp-notify: [Error running Gulp] Cannot find module 'logger' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
[04:12:47] Browserify compile error: 
 Error: Cannot find module 'parse' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:46:17
    at process (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:173:43)
    at ondir (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:188:17)
    at load (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:69:43)
    at onex (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:92:31)
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15) 

[04:12:47] gulp-notify: [Error running Gulp] Cannot find module 'parse' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
[04:12:47] Browserify compile error: 
 Error: Cannot find module 'optimize' from '/var/www/sites/cesium/cs2/node_modules/requirejs/bin'
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:46:17
    at process (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:173:43)
    at ondir (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:188:17)
    at load (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:69:43)
    at onex (/var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:92:31)
    at /var/www/sites/cesium/cs2/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)

...
```

Ideas?

**Comment by OnkelTem** [comment]
After some research the task can be rephrased to: "No support for CommonJS?".

**Comment by hpinkos** [comment]
See this issue for some info: #2981

In the future, if you have questions like this please ask them on the forum: https://groups.google.com/forum/?hl=en#!forum/cesium-dev

**Comment by OnkelTem** [comment]
@hpinkos so you've referenced a closed issue, huh? Nice work. How it is supposed to resolved it now?

**Comment by OnkelTem** [comment]
@hpinkos hello?

**Comment by hpinkos** [comment]
Hi @OnkelTem,

Sorry, I'm not very familiar with Browserify/CommonJS, but I believe the discussion in #2981 will help you solve the problems you are having.

If you don't find that helpful, I would suggested posting a question in the [forum](https://groups.google.com/forum/?hl=en#!forum/cesium-dev).  Many more users follow the forum, so there is a much better chance someone with experience using browserify will see your question and be able to help you.  Thanks!

**Comment by mramato** [comment]
@OnkelTem to use Cesium's AMD modules with browserify, you can use the [deamdify](https://www.npmjs.com/package/deamdify) plugin.  While I haven't used it personally, I know there are some people on the forum using deamdify, so if you run into problems, posting there as @hpinkos suggested

The issue @hpinkos linked to was originally about browserify and webpack support and was closed once the webpack tutorial (https://cesiumjs.org/2016/01/26/Cesium-and-Webpack/) went live.

**Comment by OnkelTem** [comment]
@mramato 

> to use Cesium's AMD modules with browserify, you can use the deamdify plugin

Thank you very much I'll try.

> The issue @hpinkos linked to was originally about browserify and webpack support and was closed once the webpack tutorial (https://cesiumjs.org/2016/01/26/Cesium-and-Webpack/) went live.

So that's the reason - the issue referenced has been closed not resolving  the _browserify part_ and so I've created this issue. (Also, there is no any mentions of deamdify in that discussion).

As for the webpack guide - yeah, I've reviewed it and while I'm not very familiar with webpack, the guide looks pretty complicated. Seems like some Cesium distribution needs some standardization/unification (judging by comparison with other libraries and how they integrate into projects).

That said, CommonJS seems to me more appropriate for a distribution :)
